### PR TITLE
Updated create_index

### PR DESCRIPTION
**Pull request by writinwaters**
### What problem does this PR solve?

Issue link:#1490

### Type of change

- [x] Documentation Update

### CODE REVIEW

File: docs/references/pysdk_api_reference.md [hunk 1]
```diff
@@ -571,98 +571,162 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+  - `EMVB`: An EMVB index. Works with tensors only.
+  - `FullText`: A full-text index.  
+  - `IVFFlat`: An IVFFlat index. 
+  - `Secondary`: A secondary index. Works with structured data only. 
+  - `BMP`: A bitmap index. Works with sparse vectors only. 
+- **index_param_list**: `list[InitParameter(str, str)]`  
+  A list of `InitParameter` objects specifying parameter settings for the chosen index type. Each object handles one parameter setting. To set a specific index parameter, pass the parameter name and its corresponding value as two separate strings to the `InitParameter` object: 
+  - Parameter settings for an HNSW index: 
+    - `"M"`: *Optional* - Defaults to`"16"`.
+    - `"ef_construction"`: *Optional* - Defauls to`"50"`.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:609]: Defauls

File: docs/references/pysdk_api_reference.md [hunk 2]
```diff
@@ -571,98 +571,162 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+    - `"ef"`: *Optional* - Defaults to `"50"`. 
+    - `"metric"` *Required* - The distance metric to use in similarity search.
+      - `"ip"`: Inner product.
+      - `"l2"`: Euclidean distance.
+      - `"cosine"`: Cosine similarity. 
+    - `"encode"`: *Optional*
+      - `"plain"`: (Default) Plain encoding. 
+      - `"lvq"`: Locally-adaptive vector quantization.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:617]: lvq only work for float vector element data type.

File: docs/references/pysdk_api_reference.md [hunk 3]
```diff
@@ -571,98 +571,162 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+    - `"pq_subspace_bits"`: *Optional* - Defaults to "8". 
+  - Parameter settings for a full text index: 
+    - `"ANALYZER"`: *Optional* - Defaults to `"standard"`
+  - Parameter settings for an IVFFlat index:  
+    - `"centroids_count"`: *Optional* - Defaults to`"128"` 
+    - `"metric"`: *Required - The distance metric to use in similarity search.
+      - `"ip"`: Inner product.
+      - `"l2"`: Euclidean distance.
+      - `"cosine"`: Cosine similarity. 
+  - Parameter settings for a secondary index:
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:629]: No parameter.

File: docs/references/pysdk_api_reference.md [hunk 4]
```diff
@@ -571,98 +571,162 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+    - `"ANALYZER"`: *Optional* - Defaults to `"standard"`
+  - Parameter settings for an IVFFlat index:  
+    - `"centroids_count"`: *Optional* - Defaults to`"128"` 
+    - `"metric"`: *Required - The distance metric to use in similarity search.
+      - `"ip"`: Inner product.
+      - `"l2"`: Euclidean distance.
+      - `"cosine"`: Cosine similarity. 
+  - Parameter settings for a secondary index: 
+  - Parameter settings for a BMP index: 
+    - `block_size=1~256`: *Required* - The size of the block in a bitmap index. Range: `"1"` ~ `"256"`. Defaults to 16.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:631]: block_size=1~256 ?

File: docs/references/pysdk_api_reference.md [hunk 5]
```diff
@@ -571,98 +571,162 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
-table_obj = db_obj.get_table("test_index_fulltext")
+### Returns
+
+- Success: `True`
+- Failure: `Exception`
+
+### Examples
+
+```python
+# Create a table named "test_index_ivfflat" with a vector column "c1"
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:672]: from ... import

File: docs/references/pysdk_api_reference.md [hunk 6]
```diff
@@ -571,98 +571,162 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+  - Parameter settings for a secondary index: 
+  - Parameter settings for a BMP index: 
+    - `block_size=1~256`: *Required* - The size of the block in a bitmap index. Range: `"1"` ~ `"256"`. Defaults to 16.
+    - `"compress_type"`:
+      - `"compress"`: (Default) Store the block max is stored in the sparse format. Works best with small block size situations.
+      - `"raw"`: 
 
-### Examples
+:::tip NOTE
+You must import the `infinity.index` package to set `IndexType`:
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:637]: You must import => You may import

File: docs/references/pysdk_api_reference.md [hunk 7]
```diff
@@ -571,98 +571,162 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+    - `"ef_construction"`: *Optional* - Defauls to`"50"`.
+    - `"ef"`: *Optional* - Defaults to `"50"`. 
+    - `"metric"` *Required* - The distance metric to use in similarity search.
+      - `"ip"`: Inner product.
+      - `"l2"`: Euclidean distance.
+      - `"cosine"`: Cosine similarity. 
+    - `"encode"`: *Optional*
+      - `"plain"`: (Default) Plain encoding. 
+      - `"lvq"`: Locally-adaptive vector quantization.
+  - Parameter settings for an EMVB index:
```
Comment (small-turtle-1) [inline docs/references/pysdk_api_reference.md:618]: required

File: docs/references/pysdk_api_reference.md [hunk 8]
```diff
@@ -571,98 +572,163 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+    - `"ef"`: *Optional* - Defaults to `"50"`. 
+    - `"metric"` *Required* - The distance metric to use in similarity search.
+      - `"ip"`: Inner product.
+      - `"l2"`: Euclidean distance.
+      - `"cosine"`: Cosine similarity. 
+    - `"encode"`: *Optional*
+      - `"plain"`: (Default) Plain encoding. 
+      - `"lvq"`: Locally-adaptive vector quantization. Works with float vector element only.  
+  - Parameter settings for an EMVB index: 
+    - `"pq_subspace_num"`: *Required* - Recommended: "32".
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:620]: Why recommanded.

File: docs/references/pysdk_api_reference.md [hunk 9]
```diff
@@ -571,98 +572,163 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+      - `"l2"`: Euclidean distance.
+      - `"cosine"`: Cosine similarity. 
+    - `"encode"`: *Optional*
+      - `"plain"`: (Default) Plain encoding. 
+      - `"lvq"`: Locally-adaptive vector quantization. Works with float vector element only.  
+  - Parameter settings for an EMVB index: 
+    - `"pq_subspace_num"`: *Required* - Recommended: "32". 
+    - `"pq_subspace_bits"`: *Required* - Recommended: "8". 
+  - Parameter settings for a full text index: 
+    - `"ANALYZER"`: *Optional* - Defaults to `"standard"`
```
[the remark this answers is not in the document]
Comment (JinHai-CN) [reply docs/references/pysdk_api_reference.md:623]: All parameters should be case-insensitive.

File: docs/references/pysdk_api_reference.md [hunk 10]
```diff
@@ -571,98 +572,163 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
 
-- Success: `True`
-- Failure: `Exception`
+An `IndexInfo` structure contains three fields,`column_name`, `index_type`, and `index_param_list`.
+    
+- **column_name**: `str`, *Required*  
+  The name of the column to build index on. Must not be empty. 
+- **index_type**: `IndexType`, *Required*  
+  Index type. You may want to import `infinity.index` to set `IndexType`: `from infinity.index import IndexType`  
+  - `Hnsw`: A HNSW index.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:600]: HNSW

File: docs/references/pysdk_api_reference.md [hunk 11]
```diff
@@ -571,98 +572,163 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+                index.InitParameter("metric", "l2")
+            ]
+        )
+    ],
+    None
+)
+```
+
+```python
+# Create a table named "test_index_hnsw" with a vector column "c1"
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:694]: from

File: docs/references/pysdk_api_reference.md [hunk 12]
```diff
@@ -571,98 +572,163 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+                index.InitParameter("metric", "l2")
+            ]
+        )
+    ],
+    None
+)
+```
+
+```python
+# Create a table named "test_index_fulltext" with a vector column "c1"
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:716]: from

File: docs/references/pysdk_api_reference.md [hunk 13]
```diff
@@ -571,98 +572,163 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
                               ], None)
 ```
 
 ```python
-db_obj.create_table(
-            "test_index_bmp", {
-                "c1": {"type": "sparse,30000,float,int16"}
-            }, None)
-table_obj = db_obj.get_table("test_index_bmp")
+# Create a table named "test_index_bmp" with a sparse vector column "c1"
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:727]: from

File: docs/references/pysdk_api_reference.md [hunk 14]
```diff
@@ -571,98 +572,163 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+    ],
+    None
+)
+```
+
+```python
+# Create a table named "test_index_fulltext" with a vector column "c1"
+table_obj = db_obj.create_table("test_index_fulltext", {"doctitle": {"type": "varchar"}, "docdate": {"type": "varchar"}, "body": {"type": "varchar"}}, None)
+# Create a full-text index named "my_index" on column "c1"
 table_obj.create_index("my_index",
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:719]: format

File: docs/references/pysdk_api_reference.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (writinwaters) [reply docs/references/pysdk_api_reference.md:623]: These parameters are case-sensitive. @JinHai-CN

File: docs/references/pysdk_api_reference.md (hunk 10 above)
Comment (writinwaters) [reply docs/references/pysdk_api_reference.md:600]: Hnsw

File: docs/references/pysdk_api_reference.md [hunk 15]
```diff
@@ -571,98 +572,163 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
+  - `Hnsw`: A HNSW index. 
+  - `EMVB`: An EMVB index. Works with tensors only.
+  - `FullText`: A full-text index.  
+  - `IVFFlat`: An IVFFlat index. 
+  - `Secondary`: A secondary index. Works with structured data only. 
+  - `BMP`: A Block-Max Pruning index. Works with sparse vectors only. 
+- **index_param_list**: `list[InitParameter(str, str)]`  
+  A list of `InitParameter` objects specifying parameter settings for the chosen index type. Each object handles one parameter setting. To set a specific index parameter, pass the parameter name and its corresponding value as two separate strings to the `InitParameter` object: 
+  - Parameter settings for an HNSW index: 
+    - `"M"`: *Optional* - Defaults to`"16"`.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:609]: Case insensitive

File: docs/references/pysdk_api_reference.md [hunk 16]
```diff
@@ -571,102 +572,189 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
...
 ```
+:::
+
+:::tip NOTE
+If `ConflictType` is not set, it defaults to `Error`.
+:::
+
+### Returns
+
+- Success: An index object.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:674]: Return value, should have error message

**Review by JinHai-CN — APPROVED** [review]

**Pull request merged**